core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java

```diff
@@ -647,6 +647,33 @@ public Set<String> getNamespaceNames() throws ExecutionException {
     }
   }
 
+  @Override
+  public void upgrade(Map<String, String> options) throws ExecutionException {
+    try {
+      if (!tableMetadataContainerExists()) {
+        return;
```

Comment:
Just quick silly questions.
- It's impossible to create the table metadata container if not exists?
- If not possible, don't need to throw an exception?

Reply:
There may be a misunderstanding here.
This check verifies if the table metadata container does not exist, if true this means there is no table installed, so there is no namespace to import to the namespaces metadata table. 
A limitation is that if there was a namespace that did not contain a ScalarDB-managed table, we wouldn't import it to the namespace table. It's because we assume such namespace are storage system namespaces or are not used for ScalarDB.

Reply:
Does this answer your question ?

Reply:
Thank you! Understood 👍